src/routes.js

```diff
@@ -0,0 +1,24 @@
+import {createWebHistory,createRouter} from 'vue-router'
+import ProductListing from './components/ProductListing.vue';
+import ProductDetail from './components/ProductDetail.vue';
+
+
+const routes=[
+    {
+        name: 'ProductListing',
+        path:'/',
+        component: ProductListing
+    },
+    {
+        name: 'ProductDetail',
+        path:'/productdetail/:id',
+        component: ProductDetail
+    },
+   
+];
+const router= createRouter({
+    history: createWebHistory(),
+    routes
+})
+
+export default router;
```

Comment:
https://stackoverflow.com/questions/44704968/visual-studio-code-insert-newline-at-the-end-of-files

use this setiings  to add auto new line in every file ending

Reply:
done it